dpgen/generator/model_devi.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Iterator, List, TYPE_CHECKING, Tuple, Iterator
+from dpgen.plugin import Plugin
+
+if TYPE_CHECKING:
+    import numpy as np
+    import dpdata
+
+class ModelDeviEngien(ABC):
+    """This is the base class of the model deviation engien."""
+
+    __ModelDeviPlugin=Plugin()
+
+    @staticmethod
+    def register(key):
+        return ModelDeviEngien.__ModelDeviPlugin.register(key)
+
+    @staticmethod
+    def engiens():
+        return ModelDeviEngien.__ModelDeviPlugin.plugins
+    
+    @staticmethod
+    def get_engien(key):
+        try:
+            return ModelDeviEngien.engiens()[key]
+        except IndexError as e:
+            raise RuntimeError("Unsupported engien!") from e
+
+    def __init__(self, jdata: dict, mdata: dict):
+        self.jdata = jdata
+        self.mdata = mdata
+
+    @abstractmethod
+    def make_input(self, iter_idx:int, sys_idx: int, directory: Iterator[str], conf_name: str, models: List[str]):
+        """Make a simulation input from a directory.
+
+        The length of system should be only 1.
+        
+        Parameters
+        ----------
+        iter_idx: int
+            index of iteration
+        sys_idx: int
+            index of system
+        directory: Iterator
+            generate the working directory to run simulations
+        conf_name: str
+            file name of initial system
+        models: list[str]
+            list of models
+        """
+        raise NotImplementedError("Not implemented")
+
+    @abstractmethod
+    def get_running_parameters(self, work_path: str) -> Tuple[str, List[str]]:
```

Comment:
the type of return value might be `dict`

Reply:
Fixed.